File: BE/src/main/java/de/msg/javatraining/donationmanager/persistence/model/Donation.java
package de.msg.javatraining.donationmanager.persistence.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "donation")
public class Donation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "campaign_id")
    private Campaign campaign_id;

    @Column(nullable = false)
    private Integer amount;

    @Column(nullable = false)
    private String currency;

    @ManyToOne
    @JoinColumn(name = "creator_id")
    private User createdBy;

    @Column(nullable = false)
    private LocalDate createdDate;

    @ManyToOne
    @JoinColumn(name = "benefactor")
    private Donor benefactor;

    private Boolean approved;

    @ManyToOne
    @JoinColumn(name = "approver_id")
    private User approvedBy;

    private LocalDate approveDate;

    private String notes;
}

File: FE/src/app/donations/models/donation.ts
import {User} from "../../user/models/user";
import {Campaign} from "../../campaigns/models/campaign";
import {Donor} from "../../donor/models/donor";

export class Donation {
  constructor(public id: number,
              public campaign_id: Campaign,
              public amount: number,
              public currency: string,
              public createdBy: User,
              public createdDate: Date,
              public benefactor: Donor,
              public approved: boolean,
              public approvedBy: User,
              public approveDate: Date,
              public notes: string) {
  }
}
